SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-4028/SpatialAuthorityTestActor.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialAuthorityTestActor.h"
+#include "Components/SceneComponent.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "LoadBalancing/AbstractLBStrategy.h"
+#include "Net/UnrealNetwork.h"
+#include "SpatialFunctionalTest.h"
+#include "SpatialFunctionalTestFlowController.h"
+
+ASpatialAuthorityTestActor::ASpatialAuthorityTestActor()
+{
+	PrimaryActorTick.bCanEverTick = true;
+
+	RootComponent = CreateDefaultSubobject<USceneComponent>(FName("RootComponent"));
+}
+
+void ASpatialAuthorityTestActor::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialAuthorityTestActor, AuthorityOnBeginPlay);
+	DOREPLIFETIME(ASpatialAuthorityTestActor, AuthorityOnTick);
+}
+
+void ASpatialAuthorityTestActor::BeginPlay()
+{
+	Super::BeginPlay();
+
+	if (HasAuthority() && AuthorityOnBeginPlay == 0)
```

Comment:
If HasAuthority returns true for multiple workers, then this test could still pass as the runtime would eventually replicate the correct state out to all interested workers.
Chatting with @mironec I think this would be better if AuthorityOnBeginPlay and AuthorityOnTick weren't replicated, but just local state. Then in each step we could check that one and only one worker has ever had this property set to true. This then also validates that HasAuthority is returning false on all other workers.